sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/HiveTableScanExec.scala

```diff
@@ -164,4 +164,28 @@ case class HiveTableScanExec(
   }
 
   override def output: Seq[Attribute] = attributes
+
+  override def sameResult(plan: SparkPlan): Boolean = plan match {
+    case other: HiveTableScanExec =>
+      val thisRequestedAttributes = requestedAttributes.map(cleanExpression)
+      val otherRequestedAttributes = other.requestedAttributes.map(cleanExpression)
+
+      val result = partitionPruningPred == other.partitionPruningPred &&
+        relation.sameResult(other.relation) &&
+          thisRequestedAttributes.zip(otherRequestedAttributes)
+            .forall(p => p._1.semanticEquals(p._2))
+      result
+    case _ => false
+  }
+
+  private def cleanExpression(e: Attribute): Expression = e match {
+    case a: AttributeReference =>
+      // As the root of the expression, Alias will always take an arbitrary exprId, we need
```

Comment:
Sorry for the delay, see the pic below, `left.cleanArgs == right.cleanArgs` in defalut `sameResult` return false, because `equals` in `MetastoreRelation` compare the output(`AttributeReference`) and `exprId`s are diff. `cleanExpression` can't erase the exprId in `AttributeReference`. So I think we need to override the `sameResult` in `HiveTableScanExec` like `FileSourceScanExec`. Let me know if I don't explain clearly.
![sameresult](https://cloud.githubusercontent.com/assets/1400819/18662273/36ffe756-7f4c-11e6-8ee0-4729ddfbec58.jpg)